File: node_structure/ListNode.java
package node_structure;

public class ListNode {
    public int val;
    public ListNode next;


    public ListNode(){}

    public ListNode(int val){this.val = val;}

    public ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }

    @Override
    public String toString() {
        String ans = "[" + this.val;
        ListNode cur = this;
        while (cur != null && cur.next != null) {
            ans = ans + ", " + cur.next.val;
            cur = cur.next;
        }
        ans += "]";
        return ans;
    }

    public static ListNode fromList(int[] lst) {
        if (lst.length == 0) return null;
        ListNode ans = new ListNode(lst[0]);
        ListNode cur = ans;
        for (int i = 1; i < lst.length; ++i) {
            cur.next = new ListNode(lst[i]);
            cur = cur.next;
        }
        return ans;
    }
}

File: node_structure/ListNode.py
# -*- coding: utf-8 -*-

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

    def __str__(self):
        cur = self
        ans = []
        ans.append(cur.val)
        while cur.next:
            ans.append(cur.next.val)
            cur = cur.next
        return str(ans)

    @classmethod
    def fromList(cls, lst):
        if not lst: return None
        head = ListNode(lst[0])
        cur = head
        for val in lst[1:]:
            cur.next = ListNode(val, None)
            cur = cur.next
        return head
